common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -714,6 +714,34 @@ public int indexOf(CharSequence subString, int start) {
     }
 
     /**
+     * Searches in this string for the index of the specified char {@code ch}.
+     * The search for the char starts at the specified offset {@code start} and moves towards the end of this string.
+     *
+     * @param ch the char to find.
+     * @param start the starting offset.
+     * @return the index of the first occurrence of the specified char {@code ch} in this string,
+     * -1 if found no occurrence.
+     */
+    public int indexOf(char ch, int start) {
+        if (start < 0) {
```

Comment:
Right now forEachByte will throw an exception if start < 0, so it can be incapsulated there only by breaking this behavior. But it's current behavior is consistent with String behavior. Entity-methods are not null/index safe, static utils methods (like in StringUtils) are safe.
But the old non-static method for subString AsciiString#indexOf(CharSequence subString, int start) is safe. So, yes, I made new one consistent with what we have now with this old method.
But I would prefer to refactor my and old non-static indexOf entity methods to be non-safe - consistent with String, Apache StringUtils and common practice. WDYT @Scottmitch ?